telco-ran/configuration/kube-compare-reference/node-tuning-operator/PerformanceProfile.yaml

```diff
@@ -1,3 +1,39 @@
+{{- /* Architecture detection loop: /* -}}
+{{- /* Find the architecture of the node which matches this object's nodeSelector */ -}}
+{{- $arch := "" }}
+{{- $profileSelectors := (keys .spec.nodeSelector) }}
+{{- $nodes := lookupCRs "v1" "Node" "" "" }}
+{{- range $node := $nodes }}
+  {{- /* Skip any nodes that do not match our nodeSelector criteria */ -}}
+  {{- range $selector := $profileSelectors }}
+    {{- if not (hasKey ($node.metadata.labels | default dict) $selector) }}
```

Comment:
Does this presume the selector is always based solely on the key? ie could no check of the value here could lead to a wrong match?

Reply:
Yes.  Low likelihood for our customer base, but worth addressing.

Will be handled in https://issues.redhat.com/browse/OCPBUGS-61834